File: scripts/dump_businesses.js
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

dotenv.config({ path: path.join(__dirname, '..', '.env') });

const url = process.env.VITE_SUPABASE_URL;
const key = process.env.VITE_SUPABASE_ANON_KEY;

if (!url || !key) {
    console.error("Missing Supabase credentials");
    process.exit(1);
}

const supabase = createClient(url, key);

async function dump() {
    console.log("--- Dumping Businesses ---");
    const { data, error } = await supabase
        .from('businesses')
        .select('id, name, is_premium, tier, logo_url, created_at')
        .order('created_at', { ascending: false });

    if (error) {
        console.error(error);
        return;
    }

    data.forEach(biz => {
        console.log(`ID: ${biz.id}`);
        console.log(`Name: ${biz.name}`);
        console.log(`Is Premium: ${biz.is_premium}`);
        console.log(`Tier: ${biz.tier}`);
        console.log(`Logo: ${biz.logo_url}`);
        console.log(`Created: ${biz.created_at}`);
        console.log("------------------------------");
    });
}

dump();

File: scripts/dump_businesses.py
import os
import json
from supabase import create_client, Client
from dotenv import load_dotenv

# Load env vars
env_path = os.path.join(os.getcwd(), '..', '.env') # Path adjustment
load_dotenv(env_path)

url = os.environ.get("VITE_SUPABASE_URL")
key = os.environ.get("VITE_SUPABASE_ANON_KEY")

if not url or not key:
    print("Missing Supabase credentials in .env")
    exit(1)

supabase: Client = create_client(url, key)

def dump_businesses():
    print("--- Dumping All Businesses ---")
    res = supabase.table("businesses").select("id, name, is_premium, tier, logo_url, created_at").order("created_at", desc=True).execute()
    
    if not res.data:
        print("No businesses found.")
        return

    for biz in res.data:
        print(f"ID: {biz.get('id')}")
        print(f"Name: {biz.get('name')}")
        print(f"Is Premium: {biz.get('is_premium')}")
        print(f"Tier: {biz.get('tier')}")
        print(f"Logo: {biz.get('logo_url')}")
        print(f"Created At: {biz.get('created_at')}")
        print("-" * 30)

if __name__ == "__main__":
    dump_businesses()
